docs/how-to/address-ipfs-on-web.md

```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
 
 ### Subdomain gateway
 
-When [origin-based security](https://en.wikipedia.org/wiki/Same-origin_policy) perimeter is needed, [CIDv1](https://github.com/ipld/cid#cidv1) in Base32 ([RFC4648](https://tools.ietf.org/html/rfc4648#section-6), no padding) should be used in the subdomain:
+When [origin-based security](https://en.wikipedia.org/wiki/Same-origin_policy) perimeter is needed, [CIDv1](/concepts/content-addressing/#identifier-formats) in case-insensitive encoding such as Base32 or Base36 should be used in the subdomain:
 
-    https://<cidv1-base32>.ipfs.<gateway-host>.tld/path/to/resource
+    https://<cidv1b32>.ipfs.<gateway-host>.tld/path/to/resource
 
 Example:
 
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.dweb.link/wiki/
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.cf-ipfs.com/wiki/Vincent_van_Gogh.html
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.localhost:8080/wiki/
 
-::: tip
+::: tip Native support in go-ipfs 0.5+
 
-go-ipfs provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
+[go-ipfs](https://dist.ipfs.io/#go-ipfs) provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
 
 Learn more about daemon configuration for hosting a public gateway:
 
 - [`Gateway.PublicGateways` docs](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) for defining gateway behavior on specified hostnames
-- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) with ready to use one-liners for most common use cases
+- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gateway-recipes) with ready to use one-liners for most common use cases
 
 :::
 
-::: danger
+::: warning Known issues
+
+- Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+- DNS label length is limited to 63 characters ([RFC 1034](https://tools.ietf.org/html/rfc1034#page-7))
 
-Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
```

Comment:
Are we advising people that they should canonically tell people to use `gateway.tld/path` and that it will redirect to the subdomain or is this unrelated?

Reply:
Yes. I've included that in _CID conversion for subdomains_ section below. 
Should we make this more prominent somehow?

Reply:
The question is more about what we're telling people. Do we tell them "hey, CIDs in subdomains is a security feature but the correct way for people to talk about URLs is using paths", or say "there are two valid ways to describe URLs"?

If it's the former then we should be louder/more prominent and explain that we'd like people to use CIDv1s so that things look nicer bc we redirect localhost:8080/ipfs/{cid} to {cid}.ipfs.localhost instead of to {cid in different representation}.ipfs.localhost. If it's the latter then we don't have to do anything.

What's your feeling on this?

Reply:
@aschmahmann I think paths make it easier  to grasp the concept of "content addressing", so my intuition is that we should be thinking in therms of  "hey, CIDs in subdomains is a security feature but the correct way for people to talk about URLs is using paths". So `gateway.tld/path` remains "canonical". 

Not sure if we need to be more vocal: I believe people who care about subdomains already default to CIDv1, so the hash remains the same. What we cna do, is to ensure our GUIs copy path version instead of subdomain version etc.

Reply:
I think your approach is reasonable here as long as we're also trying to ensure that our comms focus on path domains. Sometimes subdomains will inevitably slip in as someone copy-pastes `cid.ipfs.localhost` from their URL bar and posts it somewhere, but not much to be done there at the moment.